Fix: content.php: last-modified/if-unmodified-since (#3771)

ORelio opened the pull request:
Fix #3771
* Add support for Unix timestamps in  `Last-Modified` HTTP header
* Send `If-Unmodified-Since` in IMF-fixdate format instead of Unix timestamps
* If invalid, ignore `Last-Modified` HTTP header instead of raising an exception

See also:
* https://www.rfc-editor.org/rfc/rfc9110#http.date
* https://stackoverflow.com/questions/21120882/

ORelio committed:
Fix: content.php: last-modified/if-unmodified-since (#3771)

Fix exception if server sent invalid Last-Modified header
Add support for Unix time instead of standard date string
Send back standard RFC7231 date string instead of Unix time

dvikan commented:
can you share which url gives timestamp in the `last-modified` header? it is very rare.

your pr is good except for the fact that curl expects the `CURLOPT_TIMEVALUE` to be a unix timestamp [0]:

```php
if ($config['if_not_modified_since']) {
    curl_setopt($ch, CURLOPT_TIMEVALUE, $config['if_not_modified_since']);
    curl_setopt($ch, CURLOPT_TIMECONDITION, CURL_TIMECOND_IFMODSINCE);
}
```

[0] https://www.php.net/manual/en/function.curl-setopt.php

ORelio commented:
> can you share which url gives timestamp in the `last-modified` header? it is very rare.

```
curl -v https://www.zscaler.com/sitemap_index.xml
[...]
< HTTP/2 200
< date: Thu, 19 Oct 2023 18:32:31 GMT
< content-type: text/xml; charset=utf-8
< age: 96
< cache-control: public,max-age=300
< cf-cache-status: DYNAMIC
< content-language: en
< content-security-policy: default-src 'self' <truncated for readability>
< expires: Thu, 19 Oct 2023 18:35:56 GMT
< last-modified: 1697691694
< strict-transport-security: max-age=31536000; preload
< vary: Accept-Encoding
< x-ah-environment: prod
< x-cache: MISS
< x-content-type-options: nosniff
< x-frame-options: ALLOW-FROM testmydefenses.com
< x-geo-country: FR
< x-middleware-next: 1
< x-nf-request-id: <redacted>
< x-request-id: <redacted>
< x-robots-tag: index, follow
< x-ua-compatible: IE=edge
< x-xss-protection: 1
< server: cloudflare
< cf-ray: <redacted>
[...]
```

If you think it's rare enough, I can remove support for Unix timestamps and treat that as incorrect.

> your pr is good except for the fact that curl expects the `CURLOPT_TIMEVALUE` to be a unix timestamp [0]:

Oops! cURL did not complain when I tested my code, so did not notice. I'll revert that change. Thanks for catching that!

ORelio committed:
Fix: content.php: if-unmodified-since: cURL API

Use getTimestamp() as cURL expects that and will format the If-Modified-Since header appropriately.

ORelio commented:
Reverted change for `If-Unmodified-Since` header.
